system/View/Parser.php

```diff
@@ -698,9 +690,9 @@ public function removePlugin(string $alias)
      * Converts an object to an array, respecting any
      * toArray() methods on an object.
      *
-     * @param mixed $value
+     * @param object $value
      *
-     * @return mixed
+     * @return array
```

Comment:
You would think these should be the types but this method appears to be "change objects to arrays, leave everything else". This fall-through effect might be intentional; I would like to see more research before changing the types based solely on this.

Reply:
Better `array|float|int|object|string $value`?

- When `$value` is `object` and have method `toArray` return array
- When `$value` is `object` and not have method `toArray` return array
- Or accept else $value

I already push sample test addition

Reply:
The problem is that this is called by `setData()` where the input array `$data` is not typed, and it pass _every_ value into this method. So this method needs to be able to accept `mixed` and return everything **except** objects. Something like this:
```php
 * @param object|resource|array|string|int|float|bool|null $value
 *
 * @return resource|array|string|int|float|bool|null
 */
```
Some of those wouldn't make sense to pass to `setData()` though, so maybe we can drop `resource` and `array`? @kenjis?

Reply:
At least `resource` must not be needed.

Reply:
Yes that looks good to me!

Yes, I think this could have been named much better, but we're stuck with it for now.